### PR TITLE
alternative method of testing forms with collections

### PR DESCRIPTION
**Pull request by lordelph**
As suggested in https://github.com/symfony/symfony/issues/4124#issuecomment-431594398

### CODE REVIEW

**Comment by alexislefebvre** [comment]
That looks good, I think that you should move the text of this PR before _You can remove an existing field, e.g. a tag:_ since the code is used to add field, not remove it.

File: testing.rst [hunk 1]
```diff
@@ -789,6 +789,25 @@ You can remove an existing field, e.g. a tag::
     // the tag has been removed
     $this->assertEquals(0, $crawler->filter('ul.tags > li')->count());
 
+You can also manipulate the DOM before obtaining the form - this can be useful for simulating collections. For example,
+if you had a form which could include dynamic upload fields, you can add the extra fields to the DOM before obtaining
```
Comment (OskarStark) [inline testing.rst]: DOM and obtaining is named twice, the sentence is a bit confusing

**Comment by javiereguiluz** [comment]
@xabbuh what's your opinion about this trick? Thank you.

**Comment by xabbuh** [comment]
This sounds like quite a fragile approach to me. IMO when your functional test requires JavaScript to be executed you should use a real browser in the background (for example using Panther) instead.

**Comment by alexislefebvre** [comment]
@xabbuh You're right but in the same time, this PR uses an approach that is easy to write and maintain. Maybe we should add an info message like this:

> It is recommended to use a real browser but you can use this method that doesn't require a browser.

**Comment by xabbuh** [comment]
Maybe we should add a more generic block hinting that you can modify the DOM if you want to simulate client-side modifications and then mention that this does work but is not recommended and one should use a tool like Panther instead.

**Comment by wouterj** [comment]
Hi @lordelph! Thanks for starting this PR.

As it seems like the current changes of this PR will not be merged and there hasn't been any update in a year, I'll close this PR. Feel free to comment or update the PR, we can always hit the reopen button again.

**Pull request closed without merging**